### ISSUE TITLE
Hitting Environment.FailFast when opening ProjectSystem.sln

### ISSUE THREAD

**Issue by davidwengier** [dotnet/roslyn#38603]
Not sure exactly the repro here, so hopefully the exception and dump are useful.

On VS master, 29307.11

```
Application: devenv.exe
Framework Version: v4.0.30319
Description: The application requested process termination through System.Environment.FailFast(string message).
Message: System.AggregateException: One or more errors occurred. ---> System.InvalidOperationException: Operation is not valid due to the current state of the object.
   at Microsoft.CodeAnalysis.SmallDictionary`2.HandleInsert(AvlNode node, AvlNode parent, K key, V value, Boolean add)
   at Microsoft.CodeAnalysis.SmallDictionary`2.Insert(Int32 hashCode, K key, V value, Boolean add)
   at Microsoft.CodeAnalysis.AnalyzerConfigSet.<GetOptionsForSourcePath>g__addOptions|6_0(Section section, Builder treeBuilder, Builder analyzerBuilder, ArrayBuilder`1 diagnosticBuilder, String analyzerConfigPath, SmallDictionary`2 diagIdCache)
   at Microsoft.CodeAnalysis.AnalyzerConfigSet.GetOptionsForSourcePath(String sourcePath)
   at Microsoft.CodeAnalysis.DocumentState.CreateTreeAndVersion(ValueSource`1 newTextSource, ProjectId cacheKey, String filePath, ParseOptions options, AnalyzerConfigSet analyzerConfigSet, HostLanguageServices languageServices, PreservationMode mode, TextAndVersion textAndVersion, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.DocumentState.<FullyParseTreeAsync>d__11.MoveNext()
   --- End of inner exception stack trace ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.CodeAnalysis.DocumentState.<GetSyntaxTreeAsync>d__42.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.LanguageServices.Implementation.Progression.SymbolContainment.<GetContainedSyntaxNodesAsync>d__0.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.LanguageServices.Implementation.Progression.ContainsChildrenGraphQuery.<GetGraphAsync>d__0.MoveNext()
---> (Inner Exception #0) System.InvalidOperationException: Operation is not valid due to the current state of the object.
   at Microsoft.CodeAnalysis.SmallDictionary`2.HandleInsert(AvlNode node, AvlNode parent, K key, V value, Boolean add)
   at Microsoft.CodeAnalysis.SmallDictionary`2.Insert(Int32 hashCode, K key, V value, Boolean add)
   at Microsoft.CodeAnalysis.AnalyzerConfigSet.<GetOptionsForSourcePath>g__addOptions|6_0(Section section, Builder treeBuilder, Builder analyzerBuilder, ArrayBuilder`1 diagnosticBuilder, String analyzerConfigPath, SmallDictionary`2 diagIdCache)
   at Microsoft.CodeAnalysis.AnalyzerConfigSet.GetOptionsForSourcePath(String sourcePath)
   at Microsoft.CodeAnalysis.DocumentState.CreateTreeAndVersion(ValueSource`1 newTextSource, ProjectId cacheKey, String filePath, ParseOptions options, AnalyzerConfigSet analyzerConfigSet, HostLanguageServices languageServices, PreservationMode mode, TextAndVersion textAndVersion, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.DocumentState.<FullyParseTreeAsync>d__11.MoveNext()<---

Stack:
   at System.Environment.FailFast(System.String, System.Exception)
   at Microsoft.CodeAnalysis.FailFast.OnFatalException(System.Exception)
   at Microsoft.CodeAnalysis.ErrorReporting.FatalError.Report(System.Exception, System.Action`1<System.Exception>)
   at Microsoft.CodeAnalysis.ErrorReporting.FatalError.ReportUnlessCanceled(System.Exception)
   at Roslyn.Utilities.TaskExtensions+<>c__DisplayClass6_0`1[[System.Boolean, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]].<SafeContinueWith>g__outerFunction|0(System.Threading.Tasks.Task)
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean)
   at System.Threading.Tasks.Task`1[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]].GetResultCore(Boolean)
   at System.Threading.Tasks.Task`1[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]].get_Result()
   at Microsoft.VisualStudio.LanguageServices.Implementation.Progression.GraphQueryManager+<>c__DisplayClass14_0.<PopulateContextGraphAsync>b__1(System.Threading.Tasks.Task`1<Microsoft.VisualStudio.LanguageServices.Implementation.Progression.GraphBuilder[]>)
   at Roslyn.Utilities.TaskExtensions+<>c__DisplayClass5_0`1[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]].<SafeContinueWith>b__0(System.Threading.Tasks.Task)
   at Roslyn.Utilities.TaskExtensions+<>c__DisplayClass2_0.<SafeContinueWith>g__continuationFunction|0(System.Threading.Tasks.Task)
   at Roslyn.Utilities.TaskExtensions+<>c__DisplayClass6_0`1[[System.Boolean, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]].<SafeContinueWith>g__outerFunction|0(System.Threading.Tasks.Task)
   at System.Threading.Tasks.ContinuationResultTaskFromTask`1[[System.Boolean, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]].InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
   at System.Threading.Tasks.Task.ExecutionContextCallback(System.Object)
   at System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(System.Threading.Tasks.Task ByRef)
   at System.Threading.Tasks.Task.ExecuteEntry(Boolean)
   at System.Threading.Tasks.Task.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
   at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback()
```

Dump at `\\mlangfs1\public\dawengie\ProjectSystemCrashOnStartup.zip`.

**Comment by jinujoseph** [comment]
cc @AndyGerlicher

**Comment by jaredpar** [comment]
This is fixed by #38592

No issue was originally filed as the report came over teams. This is the stack trace we fixed though. 

CC @agocke
